### PR TITLE
drivers: sensor: adxl345: Update driver to use i2c_dt_spec

### PR DESCRIPTION
**Pull request by benjaminbjornsson**
Simplify driver by using i2c_dt_spec for bus access.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

### CODE REVIEW

**Review by gmarull — DISMISSED** [review]

File: drivers/sensor/adxl345/adxl345.h [hunk 1]
```diff
@@ -52,8 +50,7 @@ struct adxl345_sample {
 };
 
 struct adxl345_dev_config {
-	const char *i2c_master_name;
-	uint16_t i2c_addr;
+	struct i2c_dt_spec i2c;
```
Comment (galak) [inline drivers/sensor/adxl345/adxl345.h:53]: make it `const`

File: drivers/sensor/adxl345/adxl345.c [hunk 2]
```diff
@@ -20,12 +20,11 @@ LOG_MODULE_REGISTER(ADXL345, CONFIG_SENSOR_LOG_LEVEL);
 static int adxl345_read_sample(const struct device *dev,
 			       struct adxl345_sample *sample)
 {
-	struct adxl345_dev_data *data = dev->data;
+	struct adxl345_dev_config *cfg = dev->config;
```
Comment (galak) [inline drivers/sensor/adxl345/adxl345.c:23]: should be `const struct...` (same in other places)

**Review by galak — DISMISSED** [review]
Looks good, a few minor issues with usage of `const`

File: drivers/sensor/adxl345/adxl345.h [hunk 3]
```diff
@@ -52,8 +50,7 @@ struct adxl345_sample {
 };
 
 struct adxl345_dev_config {
-	const char *i2c_master_name;
-	uint16_t i2c_addr;
+	const struct i2c_dt_spec i2c;
```
Comment (gmarull) [inline drivers/sensor/adxl345/adxl345.h:53]:
```suggestion
	struct i2c_dt_spec i2c;
```

File: drivers/sensor/adxl345/adxl345.c [hunk 4]
```diff
@@ -20,12 +20,11 @@ LOG_MODULE_REGISTER(ADXL345, CONFIG_SENSOR_LOG_LEVEL);
...
 {
-	struct adxl345_dev_data *data = dev->data;
+	const struct adxl345_dev_config *cfg = dev->config;
 	int16_t raw_x, raw_y, raw_z;
 	uint8_t axis_data[6];
 
-	int rc = i2c_burst_read(data->i2c_master,
-				data->i2c_addr,
+	int rc = i2c_burst_read_dt(&cfg->i2c,
 				ADXL345_X_AXIS_DATA_0_REG,
```
Comment (gmarull) [inline drivers/sensor/adxl345/adxl345.c:28]: please fix alignment (it was before)

**Review by gmarull — CHANGES_REQUESTED** [review]

File: drivers/sensor/adxl345/adxl345.h (hunk 3 above)
Comment (benjaminbjornsson) [reply drivers/sensor/adxl345/adxl345.h:53]: > Suggested change 
> 	const struct i2c_dt_spec i2c;
> 	struct i2c_dt_spec i2c;

@galak and @gmarull, you seem to disagree on this, which one should we go with?

**Review by gmarull — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Comment by benjaminbjornsson** [comment]
@carlescufi merge ping

**Pull request merged**